File: paper/rust/bug1707931/src/main.c
#include <stdio.h>

int main(void) {
    unsigned int u = 42;
    if (u < 0)
        printf("Should never happen");

    int i = -42;
    unsigned int u2 = i;
    if (u2 < 0)
        printf("Should never happen");

    return 0;
}

File: paper/rust/bug1707931/src/main.rs
fn main() {
    let u = 42u32;
    if u < 0 {
        println!("Should never happen");
    }

    let i = -42i32;
    let u:u32 = i;
    if u < 0 {
        println!("Should never happen");
    }
}
